scripts/storage-emulator-integration/tests.ts

```diff
@@ -2111,121 +1877,171 @@ describe("Storage emulator", () => {
         });
       });
     });
+  });
 
-    emulatorSpecificDescribe("Non-SDK Endpoints", () => {
-      beforeEach(async () => {
-        await resetStorageEmulator(STORAGE_EMULATOR_HOST);
+  describe("Non-SDK Endpoints", () => {
+    const filename = "testing/storage_ref/image.png";
+    const encoded_filename = "testing%2Fstorage_ref%2Fimage.png";
+    const image_filename = writeToFile(
+      "image_base64",
+      Buffer.from(IMAGE_FILE_BASE64, "base64"),
+      tmpDir
+    );
 
-        await page.evaluate(
-          (IMAGE_FILE_BASE64, filename) => {
-            const auth = (window as any).auth as firebase.auth.Auth;
+    beforeEach(async function (this) {
+      this.timeout(TEST_SETUP_TIMEOUT);
+      await resetEmulatorState();
```

Comment:
I know it was like this before, but I don't think we need to reset the emulator state every time here right? Some tests like generating a new token with invalid values will fail regardless of the emulator state. Any way we can make it so we're not resetting every time and only when it's needed?

I think this might be one of our time sinks when running tests but I'm not 100% sure

Reply:
discussed offline, resetting local state is very fast, resetting prod state may take longer but is worth it for maintaining a purely hermetic testing environment